scripts/oe-env-prepare.ps1

```diff
@@ -0,0 +1,446 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+$ErrorActionPreference = "Stop"
+
+$AZUREDATA_DIRECTORY = Join-Path ${env:SystemDrive} "AzureData"
+$AZUREDATA_BIN_DIRECTORY = Join-Path $AZUREDATA_DIRECTORY "bin"
+$PACKAGES_DIRECTORY = Join-Path $env:TEMP "packages"
+
+$PACKAGES = @{
+    "git" = @{
+        "url" = "https://github.com/git-for-windows/git/releases/download/v2.20.1.windows.1/Git-2.20.1-64-bit.exe"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "Git-2.20.1-64-bit.exe"
+    }
+    "7z" = @{
+        "url" = "https://www.7-zip.org/a/7z1806-x64.msi"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "7z1806-x64.msi"
+    }
+    "vs_buildtools" = @{
+        "url" = "https://aka.ms/vs/15/release/vs_buildtools.exe"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "vs_buildtools.exe"
+    }
+    "ocaml" = @{
+        "url" = "http://www.ocamlpro.com/pub/ocpwin/ocpwin-builds/ocpwin64/20160113/ocpwin64-20160113-4.02.1+ocp1-msvc64.zip"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "ocpwin64.zip"
+    }
+    "clang7" = @{
+        "url" = "http://releases.llvm.org/7.0.1/LLVM-7.0.1-win64.exe"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "LLVM-7.0.1-win64.exe"
+    }
+    "sgx_drivers" = @{
+        "url" = "http://download.windowsupdate.com/d/msdownload/update/driver/drvs/2018/01/af564f2c-2bc5-43be-a863-437a5a0008cb_61e7ba0c2e17c87caf4d5d3cdf1f35f6be462b38.cab"
+        "local_file" = Join-Path $PACKAGES_DIRECTORY "sgx_base.cab"
+    }
+    "psw" = @{
+        "url" = "https://oejenkins.blob.core.windows.net/oejenkins/Intel_SGX_PSW_for_Windows_v2.3.100.49777.exe"
```

Comment:
Is https://oejenkins.blob.core.windows.net expected to be a publicly accessible endpoint for these packages? The expectation is that 3rd party devs will use this script to set this up on their local device as well, as documented in GettingStarted.Windows.md.

We should probably build parameterization of source paths into the script explicitly so Ansible and pull from our server caches but users get the default public endpoints.

Reply:
> Is https://oejenkins.blob.core.windows.net expected to be a publicly accessible endpoint for these packages? The expectation is that 3rd party devs will use this script to set this up on their local device as well, as documented in GettingStarted.Windows.md.

@CodeMonkeyLeet Yes, the https://oejenkins.blob.core.windows.net endpoint comes from our OE Jenkins storage account, and the `oejenkins` container is public, so everything starting with https://oejenkins.blob.core.windows.net/oejenkins can be publicly accessible without any problem.

The Intel PSW installer is uploaded in our storage account on purpose, for caching purposes, in case Intel removes the installer from their web server (we encountered something like this with issue https://github.com/microsoft/openenclave/issues/1617).

> 
> We should probably build parameterization of source paths into the script explicitly so Ansible and pull from our server caches but users get the default public endpoints.

Instead of hard-coding the download URLs into the scripts itself, are you suggesting to have script parameters for every download URL ? I think this would be a good idea in case somebody uses the script and caches the script installer URLs into their internal cache web server.

Reply:
Yes, I am suggesting that the download URLs be parameterized inputs into the script. Our ansible use of it should provide our blob storage endpoint, and the script can default to a set of public endpoints from Intel for public users. I do not want us to serve our cache the the broader internet.

Reply:
@CodeMonkeyLeet 

This sounds good. We'll be doing this. We have public endpoints for every file downloaded from the internet, except this one:
```
$AzureDCAPNupkgURL = 'https://oejenkins.blob.core.windows.net/oejenkins/Microsoft.Azure.DCAP.Client.1.0.0.nupkg'
```

The `Microsoft.Azure.DCAP.Client.1.0.0.nupkg` is a custom nupkg file with the `v1.0` release from [Microsoft/Azure-DCAP-Client](https://github.com/Microsoft/Azure-DCAP-Client), hosted in our storage account.

Do you know if there is an official public URL for this nupkg file ?

Ideally, I believe that this one should be attached to the official GitHub releases https://github.com/Microsoft/Azure-DCAP-Client/releases, and we can have the GitHub URL as default download URL in the script. But right now, only an archive with the library built is attached there.

Reply:
Added official download links for all software except ```Microsoft.Azure.DCAP.Client.1.0.0.nupkg```

Reply:
@ionutbalutoiu Yes, I agree that it should be pulled from their GitHub releases.

@soccerGB @anitagov Can we work with the owners of Azure DCAP client to publish a copy of their client nupkg from their GitHub? I understand that we've contributed to help stabilize their package drop for Windows.

Reply:
@CodeMonkeyLeet @soccerGB @anitagov 

I created the issue https://github.com/Microsoft/Azure-DCAP-Client/issues/74 to have this tracked.

In my opinion, we can merge this pull request with our internal storage account download link, and add a `TODO` comment in order to have this updated later with the official download link.

I'll make sure I follow up later and address the `TODO` once the `nupkg` file is published with the Azure-DCAP-Client releases.